src/ir/function.rs

```diff
@@ -309,6 +309,31 @@ pub fn cursor_mangling(
     Some(mangling)
 }
 
+fn args_from_ty_and_cursor(
+    ty: &clang::Type,
+    cursor: &clang::Cursor,
+    ctx: &mut BindgenContext,
+) -> Vec<(Option<String>, TypeId)> {
+    match (cursor.args(), ty.args()) {
+        (Some(cursor_args), Some(ty_args)) => {
+            ty_args.iter().enumerate().map(|(i, ty)| {
```

Comment:
So, what about the following instead, which could be much shorter:

```
let args_for_names = cursor.args().unwrap();
let ty_args = ty.args();
let ty_args = ty_args.as_ref().unwrap_or(&args_for_names);

// Write the loop only once.
```

Reply:
I tried to rewrite this a few different ways but found everything unsatisfactory. Cursor::args() and Type::args() return different types and I haven't figured out a good way in Rust to work with iterators of different concrete types well. Definitely open to suggestions!

Reply:
Meh, this doesn't look too bad as-is. Thanks again for looking into this!